src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -101,20 +101,41 @@ static string DecodeFrom64(string encodedData)
         /// <returns>Subscription addresses that were found for the provided messageTypes</returns>
         public async Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes, ContextBag context)
         {
-            var subscribers = new List<Subscriber>();
+            var subscribers = new HashSet<Subscriber>(new SubscriberComparer());
             var table = client.GetTableReference(subscriptionTableName);
 
             foreach (var messageType in messageTypes)
```

Comment:
I don't think it helps much because then you need to have multiple hashsets and merge them together or use a "concurrent hash set". I'd leave it as is